shared-functions.sh

```diff
@@ -80,16 +80,15 @@ ensure_mac_os() {
 
 # Mac-specific function to install Java JDK
 install_mac_java() {
-    # Determine which Java JDKs we have.
-    # If a non-Adopt Open JDK that is v1.11 (aka Java 11), that's ok!
-    java_versions=$(/usr/libexec/java_home --version "1.11" >/dev/null 2>&1 || echo "Not found")
-
-    if [ "$java_versions" = "Not found" ]; then
-        echo "Installing Adopt Open JDK v11..."
-        brew install homebrew/cask-versions/adoptopenjdk11
-    else
-        echo "java11 already installed ($java_versions)"
+    # It's suprisingly difficult to tell what java versions are
+    # already installed -- there are different java providers -- so we
+    # just always try to install the one we want.  For more info, see
+    #   https://github.com/Khan/khan-dotfiles/pull/61/files#r964917242
+    echo "Installing Adopt Open JDK v11..."
+    if ! brew tap | grep -qi "AdoptOpenJDK/openjdk"; then
```

Comment:
Almost there!!  Good thinking making grep case insensitive...  We can remove the `q` and this works like a charm!